dsc_lib/src/dscresources/command_resource.rs

```diff
@@ -583,62 +573,110 @@ pub fn invoke_command(executable: &str, args: Option<Vec<String>>, input: Option
     if let Some(env) = env {
         command.envs(env);
     }
-
     if executable == "dsc" && env::var("DEBUG_DSC").is_ok() {
         // remove this env var from child process as it will fail reading from keyboard to allow attaching
         command.env_remove("DEBUG_DSC");
     }
 
-    let mut child = command.spawn()?;
+    let mut child = match command.spawn() {
+        Ok(c) => c,
+        Err(e) => {
+            return Err(DscError::CommandOperation(e.to_string(), executable.to_string()))
+        }
+    };
+
+    let stdout = child.stdout.take().expect("child did not have a handle to stdout");
+    let stderr = child.stderr.take().expect("child did not have a handle to stderr");
+    let mut stdout_reader = BufReader::new(stdout).lines();
+    let mut stderr_reader = BufReader::new(stderr).lines();
+
     if let Some(input) = input {
         trace!("Writing to command STDIN: {input}");
-        // pipe to child stdin in a scope so that it is dropped before we wait
-        // otherwise the pipe isn't closed and the child process waits forever
-        let Some(mut child_stdin) = child.stdin.take() else {
-            return Err(DscError::CommandOperation("Failed to open stdin".to_string(), executable.to_string()));
-        };
-        child_stdin.write_all(input.as_bytes())?;
-        child_stdin.flush()?;
+        let mut stdin = child.stdin.take().expect("child did not have a handle to stdin");
+        stdin.write_all(input.as_bytes()).await.expect("could not write to stdin");
+        drop(stdin);
     }
 
-    let Some(mut child_stdout) = child.stdout.take() else {
-        return Err(DscError::CommandOperation("Failed to open stdout".to_string(), executable.to_string()));
+    let child_id: u32 = match child.id() {
+        Some(id) => id,
+        None => {
+            return Err(DscError::CommandOperation("Can't get child process id".to_string(), executable.to_string()));
+        }
     };
```

Comment:
Would prefer the more compact style:

```suggestion
    let Some(child_id) = child.id() else {
        return Err(DscError::CommandOperation("Can't get child process id".to_string(), executable.to_string()));
    };
```

Reply:
Updated.